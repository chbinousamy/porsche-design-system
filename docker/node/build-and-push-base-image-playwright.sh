#!/bin/bash

REGISTRY=ghcr.io/chbinousamy/porsche-design-system
IMAGE=playwright
TAG=v1.39.0-focal-test

docker pull $REGISTRY/$IMAGE:latest

docker build -f Dockerfile-playwright -t $REGISTRY/$IMAGE:$TAG -t $REGISTRY/$IMAGE:latest .

docker push $REGISTRY/$IMAGE:$TAG
docker push $REGISTRY/$IMAGE:latest
