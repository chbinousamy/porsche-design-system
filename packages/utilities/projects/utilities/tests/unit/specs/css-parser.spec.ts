import * as fs from 'node:fs';
import * as path from 'node:path';
import { jssTestCss } from './jss-test';

const cssFilePath = path.resolve(__dirname, 'generated/vanilla-extract.css');
const vanillaExtractTestCss = fs.readFileSync(cssFilePath, 'utf-8');

it('should match compiled styles with jss (styles package)', () => {
  expect(jssTestCss).toMatchSnapshot();
});

it('should match compiled styles with vanilla-extract (styles/vanilla-extract package)', () => {
  expect(vanillaExtractTestCss).toMatchSnapshot();
});

it('should have equal compiled css for jss and vanilla-extract', () => {
  // JSS adds newline at the end
  expect(vanillaExtractTestCss + '\n').toStrictEqual(jssTestCss);
});
