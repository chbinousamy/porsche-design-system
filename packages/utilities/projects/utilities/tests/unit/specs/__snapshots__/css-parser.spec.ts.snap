// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match compiled styles with jss (styles package) 1`] = `
".test {
  border-radius: 4px;
  margin-left: -2px;
  margin-right: -2px;
  padding-left: 2px;
  padding-right: 2px;
  transition: background var(--p-transition-duration, 0.25s) cubic-bezier(0.25,0.1,0.25,1);
}
.test:focus {
  outline: 2px solid #1A44EA;
  outline-offset: 2px;
}
.test:focus:not(:focus-visible) {
  outline-color: transparent;
}
@media (hover:hover) {
  .test:hover {
    background: rgba(148, 149, 152, .18);
  }
}
"
`;

exports[`should match compiled styles with vanilla-extract (styles/vanilla-extract package) 1`] = `
".test {
  border-radius: 4px;
  margin-left: -2px;
  margin-right: -2px;
  padding-left: 2px;
  padding-right: 2px;
  transition: background var(--p-transition-duration, 0.25s) cubic-bezier(0.25,0.1,0.25,1);
}
.test:focus {
  outline: 2px solid #1A44EA;
  outline-offset: 2px;
}
.test:focus:not(:focus-visible) {
  outline-color: transparent;
}
@media (hover:hover) {
  .test:hover {
    background: rgba(148, 149, 152, .18);
  }
}"
`;
